### ISSUE TITLE
Matter protocol similarities

### ISSUE THREAD

**Issue by backkem** [w3c/openscreenprotocol#308]
I was reading the Matter spec an it occurred to me how similar some of the protocol is to OSP. Matter also does mDNS discovery and secure channel using PAKE and even has a basic remote playback device type. I wonder if it makes sense to unify both efforts. Though Matter doest include a video streaming spec yet I believe.

**Comment by tidoust** [comment]
Yes, the Second Screen Working Group is looking at interoperability and possible alignment with Matter:
- Overall goal was [noted in the group charter](https://www.w3.org/2022/03/charter-secondscreen-wg.html#csa)
- @mfoltzgoogle presented a [comparison between Matter and the Open Screen Protocol](https://www.w3.org/2023/03/08-webscreens-minutes.html#t11) to the group during the March 2023 virtual meeting
- It is on [the agenda for the upcoming Second Screen working group face-to-face](https://github.com/w3c/secondscreen-wg/issues/10) at TPAC in September 2023

What the group did not have was a tracking issue, thanks for creating one @backkem!

**Comment by backkem** [comment]
Great comparison @mfoltzgoogle! sorry I didn't catch that earlier.

I wanted to share a look at discovery/rendezvous from a functional standpoint. It seems each approach is born from a slightly different use-case: 
- Matter focuses on creating an established network or 'Fabric'. This makes sense in a home setting where you connect once and get access to multiple (IoT) devices for longer time period.
- The OSP focuses more on point connections such as a meeting room with a TV you want to cast to.
- Similarly the [local-peer-to-peer](https://github.com/ibelem/local-peer-to-peer) proposal puts a specific focus on more ephemeral connections between devices that are physically in close proximity.

They all seem valid use-cases to me. I think the merit to building these on a unified stack is to foster (broader) adoption and avoid a fractured ecosystem (as embodied by [this comment](https://github.com/WICG/proposals/issues/103#issuecomment-1686596955)). Maybe there is room to contribute a lightweight connection establishment flow back to Matter based on the OSP work.

**Comment by anssiko** [comment]
@backkem, I like how you bucketed discovery requirements into "established" and "ephemeral" connections.

I'd want to refer here to the Matter browser support issue https://github.com/project-chip/connectedhomeip/issues/4270 you opened, also acknowledged in our [Second Screen WG March 2023 discussions](https://www.w3.org/2023/03/08-webscreens-minutes.html#t11). I think it is worth revising this issue with new information to see if the "future" is now. @backkem you're welcome to give the WG a brief update on your latest thinking at our upcoming meeting. This issue is now on the agenda: https://github.com/w3c/secondscreen-wg/issues/10

I'm expecting @mfoltzgoogle to give an update on a few specific technical areas of investigation for Matter and OSP interoperability. I'm also expecting Apple participants to attend which might help bridge the two efforts.

**Comment by rektide** [comment]
Amazon announced their shipping some Matter casting systems. https://www.theverge.com/2024/1/9/24030324/amazon-matter-casting-echo-show-fire-tv-prime-video

**Comment by rektide** [comment]
Amazon now has Matter casting in the Prime Video app and on their Echo Show 15 device. https://www.theverge.com/2024/1/9/24030324/amazon-matter-casting-echo-show-fire-tv-prime-video

**Comment by backkem** [comment]
Matter 1.5 has shipped initial WebRTC support (mainly for camera streaming). It comes with a standardized way to perform signaling between Matter devices, afterwards the WebRTC stack takes over as transport.

I think technically one could implement the OSP Application Protocol (or directly the Presentation API, Remote Playback API or Local Peer-to-Peer API) on top of this.
